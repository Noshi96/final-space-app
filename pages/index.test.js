/* globals describe, it, expect  */
/* eslint-env jest */

import { render, waitFor } from '@testing-library/react'
import Home from './index.page'

describe('Home', () => {
  const characters = [
    {
      id: 1,
      name: 'Gary Goodspeed',
      status: 'Alive',
      species: 'Human',
      gender: 'Male',
      hair: 'Blonde',
      origin: 'Earth',
      abilities: [
        'Piloting',
        'Marksmanship',
        'Hand-to-hand combat',
        'Weapons: Blasters',
        'Weapons: Black hole gun (briefly)',
        'Weapons: Robot arm',
        'Weapons: Laser gun',
        'Weapons: Laser sword'
      ],
      img_url:
        'https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png'
    }
  ]

  const episodes = [
    {
      id: 1,
      name: 'Chapter 1',
      air_date: '02/15/2018',
      director: 'Mike Roberts',
      writer: 'Olan Rogers',
      characters: [
        'https://finalspaceapi.com/api/v0/character/1',
        'https://finalspaceapi.com/api/v0/character/2',
        'https://finalspaceapi.com/api/v0/character/3',
        'https://finalspaceapi.com/api/v0/character/5',
        'https://finalspaceapi.com/api/v0/character/6',
        'https://finalspaceapi.com/api/v0/character/9',
        'https://finalspaceapi.com/api/v0/character/11'
      ],
      img_url: 'http://finalspaceapi.com/api/episode/image/chapter1.jpg'
    }
  ]

  it('should render Home page', async () => {
    const { container } = render(
      <Home characters={characters} episodes={episodes} />
    )
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
