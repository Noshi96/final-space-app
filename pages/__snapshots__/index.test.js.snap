// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home page 1`] = `
<div>
  <header
    class="sc-75d2f2ff-0 fbiUAK"
  >
    <div
      class="sc-75d2f2ff-1 gMhXdr"
    >
      <a
        class="sc-75d2f2ff-2 QLpgX"
        href="/"
      >
        <img
          alt="final-space-text"
          class="sc-75d2f2ff-3 fcosyw"
          src="https://images.fanart.tv/fanart/final-space-5e6df59bd745e.png"
        />
      </a>
      <nav>
        <div
          class="sc-2fc18e3d-0 aplpR"
        >
          <div />
          <div />
          <div />
        </div>
        <ul
          class="sc-3413b2c7-0 kWOcCk"
        >
          <li>
            <a
              class="sc-3413b2c7-1 cPDNgh"
              href="/characters"
            >
              Characters
            </a>
          </li>
          <li>
            <a
              class="sc-3413b2c7-1 cPDNgh"
              href="/characters"
            >
              Episodes
            </a>
          </li>
          <li>
            <a
              class="sc-3413b2c7-1 cPDNgh"
              href="/characters"
            >
              Info
            </a>
          </li>
          <li>
            <a
              class="sc-3413b2c7-1 cPDNgh"
              href="/characters"
            >
              Games
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main>
    <section
      class="sc-e79cba3d-0 fnqIxU"
    >
      <div
        class="sc-e79cba3d-1 kmsZdV"
      >
        <div
          class="sc-e79cba3d-2 gQtMgw"
        >
          <h1
            class="sc-e79cba3d-3 fRElsF"
          >
            <button>
              Elo
            </button>
            Check out your 
            <span>
              favorite
            </span>
             heroes from series and play
             
            <span>
              games
            </span>
          </h1>
          <p
            class="sc-e79cba3d-4 KpaWK"
          >
            Learn more about 
            <span>
              universum
            </span>
          </p>
          <button
            class="sc-e79cba3d-5 halYEC"
          >
            Click here
          </button>
        </div>
        <img
          alt="final-space-wallpaper"
          class="sc-e79cba3d-6 dLiPAo"
          src="https://images5.alphacoders.com/105/1055769.jpg"
        />
      </div>
    </section>
  </main>
</div>
`;
